### ISSUE TITLE
when not in screen reader mode, chat response accessibilty signal sound does not play

### ISSUE THREAD

**Issue by svdoever** [microsoft/vscode#245595]
I would like to have the option to configure a Signal Sound for when a Copilot agent is done with its work. Then I know I can stop doing push-ups:-)

**Comment by isidorn** [comment]
Sound and maybe a native notification.
fyi @joaomoreno

**Comment by svdoever** [comment]
The settings is there, but no sound is played.... still doing push-ups....

![Image](https://github.com/user-attachments/assets/c2cfb6f2-eddf-4984-98bb-c70b7a4a0ec6)

**Comment by isidorn** [comment]
This might be turned on by default for all users. But the sound would only play if the window is not focused. Also the sounds should be pleasent, and easy to understand (like an iphone notification).

fyi @roblourens

**Comment by meganrogge** [comment]
The sound currently only plays when enabled and when in screen reader optimized mode. Thus why I've labeled this a bug.

**Comment by meganrogge** [comment]
If we enable this by default, would we only play it when a user is in agent mode?

**Comment by isidorn** [comment]
Yes. Only agent mode. Only when window is not focued.
And only if the sound is pleasent and high quality

Happy to open a separate feature request if that works better.

**Comment by meganrogge** [comment]
I think we should track that separately. Do you not feel the chat response received sound is pleasant and high quality?

I would think we'd fix this bug then when the setting is `auto`, play under those conditions you stated when not in screen reader mode.

**Comment by meganrogge** [comment]
re window not focused, what if someone focuses the window, does pushups, and still wants to hear it? I think we'd need more feedback to decide that.

**Comment by svdoever** [comment]
In my usage, the windows stays focused, and I really want to know when user input is needed or the agent is done (only in agent mode), otherwise I must do really high pushups to be able to look on the screen:-) And of course also if I switched away to watch pushup instruction videos:-)

**Comment by meganrogge** [comment]
Here's the issue tracking the sound for when a user action is required #244290

**Comment by meganrogge** [comment]
This actually does work for me in most cases. The times it does not, it's hitting this issue where it's blocked because sounds have to occur within a certain time of a user's action or will be blocked. See explanation here 

https://github.com/microsoft/vscode/issues/178642#issuecomment-1840446946

**Comment by meganrogge** [comment]
To verify, without screen reader mode enabled, enable the chat response sound in settings. Make sure that you either hear the response sound when you expect (the response has come in) or logged in the console, there's a warning about playing a sound without user gesture.

**Comment by MassEast** [comment]
Just found `accessibility.signals.chatUserActionRequired` and after turning this `on` any input request from an agent now plays a sound. Sadly, I didn't find a way to make it play a sound when it finished its output (which would also be a great trigger for me).

**Comment by mgreenhough** [comment]
Simmilar issue:

"accessibility.signals.chatResponseReceived = On" - Works intermittently upon received response
"accessibility.signals.chatUserActionRequired = On" - Not functioning on either vscode_askQuestions OR terminal permission prompt.

Current extension - Cline w/ Moonshot API

This would be a really valuable functionality for me if it did work.

EDIT: I now see that this may be caused by a time threshold. Realistically, this makes the functionality unusable. #178642 addressing this was closed due to it being "not actionable". I hope this isn't the case?? It "seems" like such a simple function that provides HUGE value!

**Comment by nickchomey** [comment]
@meganrogge perhaps you could revisit this?
